OpenVote/src/com/openvote/LoginServlet.java

```diff
@@ -29,7 +34,12 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp)
 	
     public void doPost(HttpServletRequest req, HttpServletResponse resp)
             throws IOException {
-    	
+
+		UserService userService=UserServiceFactory.getUserService();
+		User user=userService.getCurrentUser();
+		
+		if (user!= null && isAdminLoggedIn()) {
```

Comment:
No, the extra function is useful because .isUserAdmin() can throw an exception and then it throws a wrench in the whole thing. The isAdminLoggedIn() takes care of it.